internal/api/grpc/user/query.go

```diff
@@ -7,18 +7,22 @@ import (
 	user_pb "github.com/zitadel/zitadel/pkg/grpc/user"
 )
 
-func UserQueriesToQuery(queries []*user_pb.SearchQuery) (_ []query.SearchQuery, err error) {
+func UserQueriesToQuery(queries []*user_pb.SearchQuery, level uint8) (_ []query.SearchQuery, err error) {
 	q := make([]query.SearchQuery, len(queries))
 	for i, query := range queries {
-		q[i], err = UserQueryToQuery(query)
+		q[i], err = UserQueryToQuery(query, level)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return q, nil
 }
 
-func UserQueryToQuery(query *user_pb.SearchQuery) (query.SearchQuery, error) {
+func UserQueryToQuery(query *user_pb.SearchQuery, level uint8) (query.SearchQuery, error) {
+	if level > 20 {
+		//can't go deeper than 20 levels of nesting.
+		return nil, errors.ThrowInvalidArgument(nil, "USER-zsQ97", "Errors.User.TooManyNestingLevels")
```

Comment:
I don't know the process to introduce new error codes, so I just eye-balled it from other examples

Reply:
The "USER-zsQ97" is alright, should be an unique ID, but the "Errors.User.TooManyNestingLevels" would also need to be translated in the different languages in https://github.com/zitadel/zitadel/blob/main/internal/static/i18n .

Reply:
I have added the English translation and copy pasted it as-is into other files since I don't have access to accurate translators https://github.com/zitadel/zitadel/pull/6361/commits/cb28328698ad81a33cc36b4611980784d5172a6c